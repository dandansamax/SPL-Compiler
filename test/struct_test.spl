struct substruct{
    int a;
};

struct aa{
    int a;
    int b;
    int c;
    struct substruct sub;
};

int main(){
    struct aa a;
    int c;
    c=a.a;
    c=a.b;
    c=a.c;
    return 0;
}