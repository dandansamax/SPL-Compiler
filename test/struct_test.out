FUNCTION main :
DEC var_0 [16]
t9 := &var_0
t7 := t9 + #12
var_1 := t7
t6 := &var_0
t4 := t6 + #8
var_1 := t4
t3 := &var_0
t1 := t3 + #4
var_1 := t1
RETURN #0
